### ISSUE TITLE
linux/arm64 docker golang install error

### ISSUE THREAD

**Issue by newro** [docker-library/golang#533]
The host PC is a linux/amd64 environment,
docker run -it --platform linux/arm64 golang and run it with
If you run go version immediately after entering the container, you will get an invalid argument error.
However, if you repeat it several times, version is output normally.
If you run go install afterward, this error occurs.

root@c471d0f3c89f:/go# go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
go: error obtaining buildID for go tool compile: fork/exec /usr/local/go/pkg/tool/linux_arm64/compile: invalid argument
go: error obtaining buildID for go tool compile: fork/exec /usr/local/go/pkg/tool/linux_arm64/compile: invalid argument
go: error obtaining buildID for go tool compile: fork/exec /usr/local/go/pkg/tool/linux_arm64/compile: invalid argument
go: error obtaining buildID for go tool compile: fork/exec /usr/local/go/pkg/tool/linux_arm64/compile: invalid argument
go: error obtaining buildID for go tool compile: fork/exec /usr/local/go/pkg/tool/linux_arm64/compile: invalid argument
go: error obtaining buildID for go tool compile: fork/exec /usr/local/go/pkg/tool/linux_arm64/compile: invalid argument
go: error obtaining buildID for go tool compile: fork/exec /usr/local/go/pkg/tool/linux_arm64/compile: invalid argument
go: error obtaining buildID for go tool compile: fork/exec /usr/local/go/pkg/tool/linux_arm64/compile: invalid argument
go: error obtaining buildID for go tool compile: fork/exec /usr/local/go/pkg/tool/linux_arm64/compile: invalid argument
go: error obtaining buildID for go tool compile: fork/exec /usr/local/go/pkg/tool/linux_arm64/compile: invalid argument
go: error obtaining buildID for go tool compile: fork/exec /usr/local/go/pkg/tool/linux_arm64/compile: invalid argument
go: error obtaining buildID for go tool compile: fork/exec /usr/local/go/pkg/tool/linux_arm64/compile: invalid argument

**Comment by tianon** [comment]
Yep, I've seen this also -- there's some new incompatibility between Go 1.23 and QEMU's user-mode emulation, and I'm not sure exactly what's the cause of it.  I get it even with just something simple like `go version` (because it tries to fork/exec for the new telemetry support).

**Comment by tianon** [comment]
It's probably worth reporting to Go upstream, but keep in mind that QEMU's user-mode emulation is something that they historically haven't ever (that I've seen) _intentionally_ supported, so interest in fixing it might be lukewarm (see https://github.com/golang/go/issues/1508, for example).

**Comment by tianon** [comment]
There's an explicit upstream report for this now: https://github.com/golang/go/issues/68976

**Comment by tianon** [comment]
This has since been fixed upstream. 👍
